pymc3/distributions/continuous.py

```diff
@@ -1621,49 +1641,23 @@ class AsymmetricLaplace(Continuous):
     --------
     `Reference <https://en.wikipedia.org/wiki/Asymmetric_Laplace_distribution>`_
     """
+    rv_op = asymmetriclaplace
 
-    def __init__(self, b, kappa, mu=0, *args, **kwargs):
-        self.b = at.as_tensor_variable(floatX(b))
-        self.kappa = at.as_tensor_variable(floatX(kappa))
-        self.mu = mu = at.as_tensor_variable(floatX(mu))
+    @classmethod
+    def dist(cls, b, kappa, mu=0, *args, **kwargs):
+        b = at.as_tensor_variable(floatX(b))
+        kappa = at.as_tensor_variable(floatX(kappa))
+        mu = mu = at.as_tensor_variable(floatX(mu))
 
-        self.mean = self.mu - (self.kappa - 1 / self.kappa) / b
-        self.variance = (1 + self.kappa ** 4) / (self.kappa ** 2 * self.b ** 2)
+        mean = mu - (kappa - 1 / kappa) / b
```

Comment:
mean and variance can be commented out for the same reason as below. They are not being used by the current codebase